File: python/7 kyu/36.py
# https://www.codewars.com/kata/58520e8edeb17c711c00019b/train/python

# Passed

def merry_christmas(funcs):
    character_function_dict = {
        func(): func.__name__ for func in funcs
    }

    string_to_form = 'Merry Christmas!'
    return ','.join([character_function_dict[char] for char in string_to_form])

def func(name, value):
    f = lambda: value
    f.__name__ = name
    return f

funcs = [func(tup[0], tup[1]) for tup in  
           [('a',   'M'),
            ('b',   'e'),
            ('c',   'r'),
            ('d',   'y'),
            ('e',   'C'),
            ('f',   'h'),
            ('g',   'i'),
            ('h',   's'),
            ('i',   't'),
            ('j',   'm'),
            ('k',   'a'),
            ('l',   ' '),
            ('m',   '!'),]]

output = merry_christmas(funcs)
print(output)

File: nodejs/7 kyu/36.js
// https://www.codewars.com/kata/5d5ee4c35162d9001af7d699/train/javascript

// Passed

function sumOfMinimums(arr) {
  const minArray = arr.map((nestedArr) =>
    nestedArr.reduce((prev, curr) => (prev < curr ? prev : curr))
  );
  return minArray.reduce((prev, curr) => prev + curr);
}

const output = sumOfMinimums([
  [1, 2, 3, 4, 5],
  [5, 6, 7, 8, 9],
  [20, 21, 34, 56, 100],
]);

console.log(output);